crates/async-std-resolver/src/net.rs

```diff
@@ -82,6 +58,38 @@ impl UdpSocket for AsyncStdUdpSocket {
     }
 }
 
+impl QuicLocalAddr for AsyncStdUdpSocket {
+    fn local_addr(&self) -> io::Result<SocketAddr> {
+        self.0.local_addr()
+    }
+}
+
+#[async_trait]
+impl UdpSocket for AsyncStdUdpSocket {
+    async fn connect(addr: SocketAddr) -> io::Result<Self> {
+        let bind_addr: SocketAddr = match addr {
+            SocketAddr::V4(_addr) => (Ipv4Addr::UNSPECIFIED, 0).into(),
+            SocketAddr::V6(_addr) => (Ipv6Addr::UNSPECIFIED, 0).into(),
+        };
+
+        Self::connect_with_bind(addr, bind_addr).await
+    }
+
+    async fn connect_with_bind(_addr: SocketAddr, bind_addr: SocketAddr) -> io::Result<Self> {
+        let socket = async_std::net::UdpSocket::bind(bind_addr).await?;
+
+        // TODO: research connect more, it appears to break receive tests on UDP
```

Comment:
I don't think we want to just leave this here?

Reply:
This is a TODO from the original code. It has little connection with my PR and I don't know that problem much. So I just leave the TODO here.

Reply:
Ah, that's fair.